internal/declcfg/load_test.go

```diff
@@ -265,9 +274,534 @@ func TestLoadDir(t *testing.T) {
 	}
 }
 
-func mustLoadFile(t *testing.T, path string) []byte {
-	t.Helper()
-	data, err := os.ReadFile(path)
-	require.NoError(t, err)
-	return data
+var (
+	invalidBundle = &fstest.MapFile{
+		Data: []byte(`{"schema": "olm.bundle","relatedImages": {}}`),
+	}
+	invalidPackage = &fstest.MapFile{
+		Data: []byte(`{"schema": "olm.package","name": {}}`),
+	}
+	noSchema = &fstest.MapFile{
+		Data: []byte(`hello: world`),
+	}
+	invalidFormat = &fstest.MapFile{
+		Data: []byte(`[This is not yaml or json.}`),
+	}
+	notObject = &fstest.MapFile{
+		Data: []byte(`[]`),
+	}
+	invalidFS = fstest.MapFS{
+		"invalid-bundle.json":  invalidBundle,
+		"invalid-package.json": invalidPackage,
+		"no-schema.yaml":       noSchema,
+		"invalid-format.txt":   invalidFormat,
+		"not-object.json":      notObject,
+	}
+
+	indexIgnore = &fstest.MapFile{
+		Data: []byte(`*
+!*.json
+!*.yaml
+
+*.clusterserviceversion.yaml`),
+	}
+	cockroachdb = &fstest.MapFile{
```

Comment:
nit: in the future, it might be nice to handcraft smaller datasets for these tests. Sharing large datasets between testcases often leads to issues later on.

Reply:
Yeah agreed. At the time, context was "let's make this look like the EP/existing tests to make sure I don't inadvertently miss something"

Now that we've got good test coverage, we could probably (at some point) refactor/minify test data and ensure coverage stays the same.